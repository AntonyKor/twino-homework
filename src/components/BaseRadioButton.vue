<template>
  <span>
    <input
      v-bind="$attrs"
      :checked="modelValue === value"
      :value="value"
      @change="$emit('change', value)"
      type="radio"
      class="radio"
    />
    <span class="checkmark"></span>
  </span>
</template>

<script>
export default {
  inheritAttrs: false,
  model: {
    prop: "modelValue",
    event: "change",
  },
  props: {
    modelValue: { type: [String, Number], default: "" },
    value: { type: [String, Number], required: true },
  },
};
</script>

<style lang="scss" scoped>
.radio {
  display: none;

  &:checked ~ .checkmark {
    border: 5px solid #2e50b6;
  }
}

.checkmark {
  content: "";
  display: inline-block;
  box-sizing: border-box;
  height: 24px;
  width: 24px;
  border: 1px solid #9e9fa2;
  border-radius: 50%;
  margin-right: 12px;
}
</style>
