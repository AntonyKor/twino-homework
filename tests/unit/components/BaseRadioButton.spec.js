import { shallowMount } from "@vue/test-utils";
import BaseRadioButton from "@/components/BaseRadioButton.vue";

describe("BaseRadioButton", () => {
  it("should match snapshot", () => {
    const wrapper = shallowMount(BaseRadioButton, {
      propsData: { value: "val" },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should pass attrs to input", () => {
    const wrapper = shallowMount(BaseRadioButton, {
      propsData: { value: "val" },
      attrs: { test: "test" },
    });

    expect(wrapper.find("input").attributes("test")).toEqual("test");
  });

  it("should on change it should emit change event with value", async () => {
    const wrapper = shallowMount(BaseRadioButton, {
      propsData: { value: "val" },
    });

    await wrapper.find("input").trigger("change");

    expect(wrapper.emitted("change")).toHaveLength(1);
    expect(wrapper.emitted("change")[0]).toEqual(["val"]);
  });
});
