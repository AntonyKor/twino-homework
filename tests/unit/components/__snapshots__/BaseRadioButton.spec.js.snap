// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseRadioButton should match snapshot 1`] = `
<span>
  <input
    class="radio"
    type="radio"
    value="val"
  />
   
  <span
    class="checkmark"
  />
</span>
`;
